pkg/inventory/inventory-client.go

```diff
@@ -441,24 +448,28 @@ func (cic *ClusterInventoryClient) updateStatus(obj *unstructured.Unstructured,
 	if !found {
 		return nil
 	}
-
-	klog.V(4).Infof("update inventory status")
 	mapping, err := cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
+	if err != nil {
+		return nil
+	}
+	hasStatus, err := cic.hasSubResource(obj.GetAPIVersion(), mapping.Resource.Resource, "status")
 	if err != nil {
 		return err
 	}
+	if !hasStatus {
+		klog.V(4).Infof("skip updating inventory status")
+		return nil
+	}
+
+	klog.V(4).Infof("update inventory status")
 	resource := cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace())
 	meta := metav1.TypeMeta{
 		Kind:       obj.GetKind(),
 		APIVersion: obj.GetAPIVersion(),
 	}
-	liveObj, err := resource.Get(context.TODO(), obj.GetName(), metav1.GetOptions{TypeMeta: meta}, "status")
+	liveObj, err := resource.Get(context.TODO(), obj.GetName(), metav1.GetOptions{TypeMeta: meta})
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.V(4).Infof("skip updating inventory subresource status for %v: %v", meta, err)
-			return nil
-		}
-		return fmt.Errorf("failed to get inventory status from cluster: %w", err)
+		return fmt.Errorf("failed to get inventory object from cluster: %w", err)
 	}
 
 	err = unstructured.SetNestedMap(liveObj.UnstructuredContent(), status, "status")
```

Comment:
UnstructuredContent is dangerous to use here. It returns an empty map if Object is nil. So if it then gets populated, the original liveObj will not be updated. 

I think in general you have to use UnstructuredContent for safe [Get]Nested calls but obj.Object for safe SetNested calls.

Reply:
`liveObj` is get from the above `Get` call. When the `Get` call doesn't return an error, the `liveObj` shouldn't be empty

Reply:
If we expect it not to be empty, then you can just use liveObj.Object, right?

Reply:
updated.